internal/pkg/api/auth.go

```diff
@@ -116,7 +116,12 @@ func authAgent(r *http.Request, id *string, bulker bulk.Bulk, c cache.Cache) (*m
 			Msg("authApiKey slow")
 	}
 
-	agent, err := findAgentByAPIKeyID(ctx, bulker, key.ID)
+	var agent *model.Agent
+	if id != nil {
+		agent, err = getAgentAndVerifyAPIKeyID(ctx, bulker, *id, key.ID)
+	} else {
+		agent, err = findAgentByAPIKeyID(ctx, bulker, key.ID)
+	}
```

Comment:
The basic approach should work, we have a 1-1 map of agents and keys because we update the keys instead of revoking+reissuing when there are changes.
We actually do the API key check below (line 151) so this method may need to be refactored to avoid redundant checks

Reply:
> We actually do the API key check below (line 151) so this method may need to be refactored to avoid redundant checks

I think we probably still want to verify api key id in a different way depending of the entry point it do not have the same meaning in one case the doc is probably corrupted in the other it's probably someone impersonating an agent.